pkg/collect/all_logs.go

```diff
@@ -0,0 +1,136 @@
+package collect
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func AllLogs(c *Collector, logsCollector *troubleshootv1beta2.AllLogs) (CollectorResult, error) {
+	client, err := kubernetes.NewForConfig(c.ClientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	output := NewResult()
+
+	ctx := context.Background()
+	// If Namespaces list is empty, get all the namespaces
+	if len(logsCollector.Namespaces) == 0 || logsCollector.Namespaces[0] == "*" {
+		_, namespaceList, namespaceErrors := getAllNamespaces(ctx, client)
+		if len(namespaceErrors) > 0 {
+			output.SaveResult(c.BundlePath, getAllLogsErrorsFileName(logsCollector), marshalErrors(namespaceErrors))
+		}
+		logsCollector.Namespaces = getNamespaceNames(namespaceList)
+	}
+
+	for _, namespace := range logsCollector.Namespaces {
+		pods, podsErrors := listPodsInNamespace(ctx, client, namespace, logsCollector.Selector)
+		if len(podsErrors) > 0 {
+			output.SaveResult(c.BundlePath, getAllLogsErrorsFileName(logsCollector), marshalErrors(podsErrors))
+		}
+
+		if len(pods) > 0 {
+			for _, pod := range pods {
+				if len(logsCollector.ContainerNames) == 0 {
+					// make a list of all the containers in the pod, so that we can get logs from all of them
+					containerNames := []string{}
+					for _, container := range pod.Spec.Containers {
+						containerNames = append(containerNames, container.Name)
+					}
+					for _, container := range pod.Spec.InitContainers {
+						containerNames = append(containerNames, container.Name)
+					}
+
+					for _, containerName := range containerNames {
+						if len(containerNames) == 1 {
+							containerName = "" // if there was only one container, use the old behavior of not including the container name in the path
+						}
+						podLogs, err := savePodLogs(ctx, c.BundlePath, client, pod, logsCollector.Name, containerName, logsCollector.Limits, false)
+						if err != nil {
+							key := fmt.Sprintf("%s/%s-errors.json", logsCollector.Name, pod.Name)
+							if containerName != "" {
+								key = fmt.Sprintf("%s/%s/%s-errors.json", logsCollector.Name, pod.Name, containerName)
+							}
+							output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
+							if err != nil {
+								return nil, err
+							}
+							continue
+						}
+						for k, v := range podLogs {
+							output[k] = v
+						}
```

Comment:
This code block is almost identical to the one bellow. Can we change the code so we're not duplicating it?